### PR TITLE
initial storm tests using locust

### PR DESCRIPTION
**Pull request by cehbrecht**
## Overview

This PR adds a "storm" test suite to check emu/pywps under load. 
It is using: 
https://locust.io/

Changes:

* Added locust tests in `tests/storm`
* Added a `tests/storm/README.md` on how to run them

### CODE REVIEW

File: tests/storm/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Run locaust tests
```
Comment (huard) [inline tests/storm/README.md:1]:
```suggestion
# Run locust tests
```

File: tests/storm/common.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+from pyquery import PyQuery
+from urllib.parse import urlparse
+import gevent
+
+
+templates_path = os.path.join(os.path.abspath(os.path.dirname(__file__)), "templates")
+template_env = Environment(loader=FileSystemLoader(templates_path), autoescape=True)
+
+
+def execute_async(client, name, identifier, inputs=None, complex_inputs=None):
```
Comment (huard) [inline tests/storm/common.py:14]: Add 1-line docstring just to clarify what each function does.

File: tests/storm/locustfile.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
+import json
+from locust import HttpUser, between, task
+
+from tests.storm.common import execute_async
+
+
+class RookUser(HttpUser):
```
Comment (huard) [inline tests/storm/locustfile.py:7]: Why the reference to Rook ?

**Review by huard — APPROVED** [review]

File: tests/storm/locustfile.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        )
+        with self.client.get(
+            query, catch_response=True, name="describe_process"
+        ) as response:
+            if "<ows:Identifier>hello</ows:Identifier>" not in response.text:
+                response.failure(
+                    "Process description for *hello* does not match expected XML"
+                )
+
+    @task(1)
```
Comment (tlvu) [inline tests/storm/locustfile.py:34]: The (1) means only 1 hit at a time?

File: tests/storm/locustfile.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+from locust import HttpUser, between, task
+
+from tests.storm.common import execute_async
+
+
+class RookUser(HttpUser):
+    host = "http://localhost:5000"
+    wait_time = between(1, 10)
+
+    @task(4)
```
Comment (tlvu) [inline tests/storm/locustfile.py:11]: (4) means 4 concurrent requests?

**Review by tlvu — APPROVED** [review]
Just questions.  Very interesting addition.

File: tests/storm/locustfile.py (hunk 5 above)
Comment (cehbrecht) [reply tests/storm/locustfile.py:11]: I'm also new to locust. The number in `task(4)` means that is 4 times more likely to be called:

Comment (cehbrecht) [reply tests/storm/locustfile.py:11]: https://docs.locust.io/en/stable/quickstart.html

File: tests/storm/locustfile.py (hunk 4 above)
Comment (cehbrecht) [reply tests/storm/locustfile.py:34]: the "1" is not necessary ... it is the default ... I will remove it.

File: tests/storm/locustfile.py (hunk 3 above)
Comment (cehbrecht) [reply tests/storm/locustfile.py:7]: copy + paste :) Changed to `EmuUser`.

File: tests/storm/README.md (hunk 1 above)
Comment (cehbrecht) [reply tests/storm/README.md:1]: fixed. Added also links to locust docs.

File: tests/storm/common.py (hunk 2 above)
Comment (cehbrecht) [reply tests/storm/common.py:14]: done.

**Comment by cehbrecht** [comment]
@tlvu @huard thanks for review :)

**Pull request merged**